robottelo/report_portal/portal.py

```diff
@@ -74,197 +80,119 @@ def _format_launches(self, launches):
-        :param str launch_type: Either satellite6 or upgrades,
-            default returns only non-upgrade launches
+        :param bool include_unfinished: Includes launches with status IN_PROGRESS or INTERRUPTED
+        :param list importances: A list of importance levels we want to fetch launches for
+        :param str name: Name of the launch to be filtered
+        :param str uuid: UUID of the launch to be fetched - overrides the other parameters
         :returns dict: The launches of Report portal.
             if sat_version is given,
             ```{'snap_version1':launch_object1, 'snap_version2':launch_object2}```
             else,
             ```{'sat_version1':{'snap_version1':launch_object1, ..}, 'sat_version2':{}}```
         """
-        if launch_type not in launch_types:
-            raise ValueError(
-                f'Wrong Launch Type is given as {launch_type} should be one of {launch_types}'
-            )
-        data = self._launch_requester()
-        sorted_launches = sorted(
-            data['content'], key=lambda launch: launch['startTime'], reverse=True
-        )
-        typed_launches = filter(
-            lambda launch: launch['name'].lower() == launch_type, sorted_launches
-        )
-        launches_ = self._format_launches(typed_launches)
-        if sat_version:
-            if sat_version not in launches_:
-                raise ValueError(
-                    f'The given satellite version \'{sat_version}\' is not available in '
-                    f'Report portal project \'{self.rp_project}\''
-                )
-            else:
-                return launches_[sat_version]
-        return launches_
-
-    def launch(self, sat_version, snap_version=None, launch_type='satellite6'):
-        """Returns a specific launch data in Report Portal Project
-
-        This does not includes each tests data in launch
-
-        :param str sat_version: The satellite version
-        :param str snap_version: The snap version of a given satellite version
-            if None, the latest launch data of a given sat_version is returned
-        :param str launch_type: Either satellite6 or upgrades, default returns only
-            non-upgrade launches
-        :returns dict: The data directory of requested or latest launch
-        """
-        launches_ = self.launches(sat_version=sat_version, launch_type=launch_type)
-        if snap_version:
-            if snap_version not in launches_.keys():
-                raise ValueError(
-                    f'The given snap_version \'{snap_version}\' is not available '
-                    f'in satellite version \'{sat_version}\''
-                )
+        # fixme: we might want to implement pagination eventually
+        if importances is None:
+            importances = self.importance_levels
+        params = {'page.page': 1, 'page.size': 500, 'page.sort': 'startTime,desc'}
```

Comment:
it's not really likely that we'll get >500 applicable launches, so I haven't implemented the pagination handling

Reply:
But this might take some toll on performance to fetch all of them in one page / one API request.

Though its not the case for us, cause we are having enough filters to fetch only one. So this count can be reduced here to 50 or even 10.

Reply:
used 20 as a reasonable compromise